docs/how-to-guides/meson-args.rst

```diff
@@ -4,77 +4,54 @@
 
 .. _how-to-guides-meson-args:
 
-***************************
-Passing arguments to Meson_
-***************************
+**************************
+Passing arguments to Meson
+**************************
 
-.. todo::
+``meson-python`` invokes the ``meson setup``, ``ninja``, and ``meson
+install`` to build the files that will be included in the Python
+wheel, and ``meson dist`` to collect the files that will be included
+in the Python sdist.
 
-   - Mention the order in which Meson arguments are added, and how it affect
-     Meson
-
-Advanced Meson options can be accessed by specifying extra arguments to the
-individual Meson commands:
-
-   - meson dist: https://mesonbuild.com/Commands.html#dist
-   - meson setup: https://mesonbuild.com/Commands.html#setup
-   - meson compile: https://mesonbuild.com/Commands.html#compile
-   - meson install: https://mesonbuild.com/Commands.html#install
-
-These arguments can be added permanently to the project by adding a section
-to the project's ``pyproject.toml``:
+Command line arguments for ``meson`` and ``ninja`` can be specified
+via tool specific settings in ``pyproject.toml`` as lists of strings
+for the ``setup``, ``compile``, ``install``, and ``dist`` keys in the
+``tool.meson-python.args`` table. For example:
 
 .. code-block:: toml
 
-	[tool.meson-python.args]
-	dist    = ['dist-argument_1', 'dist-argument_2', '...']
-	setup   = ['setup-argument_1', 'setup-argument_2', '...']
-	compile = ['compile-argument_1', 'compile-argument_2', '...']
-	install = ['install-argument_1', 'install-argument_2', '...']
-
-They can also be temporarily overwritten at build time using the
-:ref:`build config settings<how-to-guides-config-settings>`.
-
-.. tab-set::
-
-    .. tab-item:: pypa/buid
-        :sync: key_pypa_build
-
-		.. code-block:: console
-
-			$ python -m build -Cdist-args="args"    \
-					  -Csetup-args="args"   \
-					  -Ccompile-args="args" \
-					  -Cinstall-args="args" .
+   [tool.meson-python.args]
+   setup = ['-Doption=false', '-Dfeature=enable', '-Dvalue=42']
+   compile = ['-j4']
+   install = ['--tags=bindings']
+   dist = ['--include-subprojects']
 
-    .. tab-item:: pip
-        :sync: key_pip
-
-		.. code-block:: console
+Or can be specified via configuration settings passed by the Python
+build front-end as ``setup-args``, ``compile-args``, ``install-args``,
+and ``dist-args`` :ref:`config settings <how-to-guides-config-settings>`.
+Configuration settings specified via the Python build front-end have
+precedence over and override the ones specified in ``pyproject.toml``.
```

Comment:
They don't override, they can be used to override though. As-is, the sentence can read as if you pass configs via the build frontend, we will ignore the settings specified in `pyproject.toml`.

```suggestion
Configuration settings specified via the Python build front-end have
precedence over, and be used to override, the ones specified in ``pyproject.toml``.
```